From dd5789f30937ba9161c0fe4726650a5db4226164 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Mar 2017 13:23:54 +0100
Subject: [PATCH] tc/cls_flower: Add to the usage encapsulation dest UDP port

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1422629
Upstream Status: iproute2.git commit bf73c650ac93a
Conflicts: Changes applied manually due to context changes from previous
           backports.

commit bf73c650ac93ae673291c584fe592ea70686e37e
Author: Hadar Hen Zion <hadarh@mellanox.com>
Date:   Thu Dec 22 10:14:40 2016 +0200

    tc/cls_flower: Add to the usage encapsulation dest UDP port

    Encapsulation dest UDP port is part of the classifier matching
    parameters, add it to the usage.

    Fixes: 41aa17ff4668 ("tc/cls_flower: Add dest UDP port to tunnel params")
    Signed-off-by: Hadar Hen Zion <hadarh@mellanox.com>
    Reported-by: Simon Horman <simon.horman@netronome.com>
---
 tc/f_flower.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/tc/f_flower.c b/tc/f_flower.c
index 2774905..c3e0249 100644
--- a/tc/f_flower.c
+++ b/tc/f_flower.c
@@ -53,10 +53,11 @@ static void explain(void)
 		"                       dst_port PORT-NUMBER |\n"
 		"                       src_port PORT-NUMBER |\n"
 		"                       type ICMP-TYPE |\n"
-		"                       code ICMP-CODE }\n"
+		"                       code ICMP-CODE |\n"
 		"                       enc_dst_ip [ IPV4-ADDR | IPV6-ADDR ] |\n"
 		"                       enc_src_ip [ IPV4-ADDR | IPV6-ADDR ] |\n"
-		"                       enc_key_id [ KEY-ID ] }\n"
+		"                       enc_key_id [ KEY-ID ] |\n"
+		"                       enc_dst_port [ UDP-PORT ] }\n"
 		"       FILTERID := X:Y:Z\n"
 		"       MASKED_LLADDR := { LLADDR | LLADDR/MASK | LLADDR/BITS }\n"
 		"       ACTION-SPEC := ... look at individual actions\n"
-- 
1.8.3.1

